src/__test__/AbstractAppliance.test.js

```diff
@@ -0,0 +1,96 @@
+import { AssertionError } from 'assert'
+import AbstractAppliance from '../AbstractAppliance'
+import {
+	AbstractInstantiationError,
+	NotImplementedError,
+} from '../errors'
+import Payload from '../Payload'
+import {
+	FullyImplementedAppliance,
+	PartiallyImplementedAppliance,
+} from './classes'
+
+describe('AbstractAppliance #unit', () => {
+	describe('construction', () => {
+		it('should throw an error when called directly', () => {
+			expect(() => {
+				new AbstractAppliance() // eslint-disable-line no-new
+			}).toThrow(AbstractInstantiationError)
+		})
+
+		it('should throw an error when getInputTypes() is called without implementation', () => {
+			const implementedAppliance = new PartiallyImplementedAppliance()
+			expect(() => implementedAppliance.getInputTypes()).toThrow(NotImplementedError)
+		})
+
+		it('should throw an error when getOutputTypes() is called without implementation', () => {
+			const implementedAppliance = new PartiallyImplementedAppliance()
+			expect(() => implementedAppliance.getOutputTypes()).toThrow(NotImplementedError)
+		})
+
+		it('should throw an error when isValidPayload() is called without implementation', async () => {
+			const implementedAppliance = new PartiallyImplementedAppliance()
+			await expect(async () => implementedAppliance.isValidPayload())
+				.rejects.toBeInstanceOf(NotImplementedError)
+		})
+
+		it('should throw an error when invoke() is called without implementation', async () => {
+			const implementedAppliance = new PartiallyImplementedAppliance()
+			await expect(async () => implementedAppliance.invoke())
+				.rejects.toBeInstanceOf(NotImplementedError)
+		})
+
+		it('should allow construction when extended', () => {
+			expect(() => {
+				new PartiallyImplementedAppliance() // eslint-disable-line no-new
+			}).not.toThrow(Error)
+		})
+
+		it('should not throw an error when getInputTypes() is called with implementation', () => {
+			const implementedAppliance = new FullyImplementedAppliance()
```

Comment:
Same comment as above re: moving this to the outer scope and sharing it with other tests. Though of course, you'd want to rename them to be able to distinguish between the fully/partially implemented instances.

Reply:
Disregard, you convinced me [here](https://github.com/tvkitchen/appliance-core/pull/3#discussion_r418753621).